lib/app_web/components/core_components.ex

```diff
@@ -0,0 +1,621 @@
+defmodule AppWeb.CoreComponents do
+  @moduledoc """
+  Provides core UI components.
+
+  The components in this module use Tailwind CSS, a utility-first CSS framework.
+  See the [Tailwind CSS documentation](https://tailwindcss.com) to learn how to
+  customize the generated components in this module.
+
+  Icons are provided by [heroicons](https://heroicons.com), using the
+  [heroicons_elixir](https://github.com/mveytsman/heroicons_elixir) project.
+  """
+  use Phoenix.Component
+
+  alias Phoenix.LiveView.JS
+  import AppWeb.Gettext
+
+  @doc """
+  Renders a modal.
+
+  ## Examples
+
+      <.modal id="confirm-modal">
+        Are you sure?
+        <:confirm>OK</:confirm>
+        <:cancel>Cancel</:cancel>
+      </.modal>
+
+  JS commands may be passed to the `:on_cancel` and `on_confirm` attributes
+  for the caller to react to each button press, for example:
+
+      <.modal id="confirm" on_confirm={JS.push("delete")} on_cancel={JS.navigate(~p"/posts")}>
```

Comment:
**`Modals`** are _horrible_ in _most_ cases, don't use them! https://github.com/dwyl/product-ux-research/issues/38